Achimenes-freeman/rs-clone
# Rs-clone

### **Огромное спасибо всем проверяющим, кто дал нам возможность завершить проект!**
1. `Task`: https://github.com/rolling-scopes-school/tasks/blob/master/tasks/rsclone/rsclone.md
2. `Screnshots`: 
![image](https://user-images.githubusercontent.com/79699349/222426739-99b1095b-697e-4e57-be8a-7d975f56d103.png)
![image](https://user-images.githubusercontent.com/79699349/222426928-f61fe368-ba7a-4daa-b6c9-73c4341f65f3.png)
![image](https://user-images.githubusercontent.com/79699349/222427082-80e4bfb0-f641-4c2f-a399-6fc9ff22c96d.png)
![image](https://user-images.githubusercontent.com/79699349/222427171-ad36664b-83d8-4097-b702-3db723c1b618.png)
3. `Deploy`: https://achimenes-freeman.github.io/rs-clone/
4.  `Back-end`: https://github.com/grom0330/rs-clone-backend/tree/prototype
5. `Done` 28.02.2023 / `deadline` 28.08.2023
6. `Score`:  `620`
- Frontend: `470`
   * Settings: `160`
     + Test difficulty `10`
     + Quick restart bind `10`
     + Test language `10`
     + Sound volume options `10`
     + Sound on click `10`
     + Sound on error `10`
     + Make caret static or dynamic `10`
     + Style of timer `10`
     + Color of timer `10`
     + Opacity of timer `10`
     + Font size `10`
     + Font family `10`
     + Flip test colors `10`
     + Colorful mode `10`
     + Different themes `10`
     + Possibility to import/export settings(with it validation) `10`
   * Leaderboard: `20`
     + Leader is a user with a highest wpm in any game `20`
   * Test: `170`
     + Two game mods, 4 sub-mods for each `80`
     + Realtime word-check `25`
     + Realtime marking of correct/incorrect letters/words `25`
     + Realtime statistics calculation `25`
     + Possibility of sound playback `15`
     + Chart creation in the end of the test `20`
   * User account: `100`
     + Best games in each category `10`
     + Chart of the ratio of speed to distance `15`
     + User minimal statistics `10`
     + User all statistics `10`
     + Game list with game statistics `10`
     + Authentication `10`
     + Registration(with validation) `15`
     + Login(with validation) `15`
     + AuthToken saving `5`
     + Re-login when reload/reopen tab/browser `20`
- Backend `150`
   * Own backend API `15`
   * User registration `15`
   * User login `15`
   * User authToken generation `15`
   * Setting user games and their statistics `15`
   * Getting user games and their statistics `15`
   * Setting user settings `15`
   * Getting user settings `15`
   * Getting rating `15`
   * Getting user profile `15`
7. Developed by: 
   - [Achimenes-freeman](https://github.com/Achimenes-freeman) Frontend
   - [iSvitka](https://github.com/iSvitka) Frontend
   - [grom0330](https://github.com/grom0330) Backend
